microbench-ci: use no-clobber copy for builds

herkolategan opened the pull request:
Previously, the merge base build would fail on a race between other PRs building the same base SHA and contending for the copy operation. This change ensures the copy still succeeds if another job finished first.

Epic: None
Release note: None

herkolategan committed:
microbench-ci: use no-clobber copy for builds

Previously, the merge base build would fail on a race between other PRs building
the same base SHA and contending for the copy operation. This change ensures the
copy still succeeds if another job finished first.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/141482)
<!-- Reviewable:end -->

nameisbhaskar reviewed (approved):
LGTM!

herkolategan committed:
microbench-ci: explicit workflow permissions

Previously, the microbench workflow did not explicitly define permissions at a
top-level. This limits the scope of permissions.

Epic: None
Release note: None

herkolategan commented:
TFTR!

bors r=nameisbhaskar
